cluster-autoscaler/processors/capacitybuffer/scale_up_status_processor_test.go

```diff
@@ -76,6 +90,208 @@ func TestProcess(t *testing.T) {
+					{
+						Pod:                createPod("fake-pod-1", true),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+				},
+				PodsAwaitEvaluation: []*apiv1.Pod{},
+			},
+			expectedTriggeredScaleUp:    0,
+			expectedNotTriggeredScaleUp: 1,
+		},
+		"One fake pod, unschedulalble with error": {
+			buffersRegistry: NewCapacityBuffersFakePodsRegistry(map[string]*v1alpha1.CapacityBuffer{"fake-pod-1": buffer1}),
+			state: &status.ScaleUpStatus{
+				Result:               status.ScaleUpError,
+				ScaleUpInfos:         []nodegroupset.ScaleUpInfo{},
+				PodsTriggeredScaleUp: []*apiv1.Pod{createPod("pod-1", false)},
+				PodsRemainUnschedulable: []status.NoScaleUpInfo{
+					{
+						Pod:                createPod("fake-pod-1", true),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+				},
+				PodsAwaitEvaluation: []*apiv1.Pod{},
+			},
+			expectedTriggeredScaleUp:    0,
+			expectedNotTriggeredScaleUp: 0,
+		},
+		"two fake pods for same buffer, one triggers scale up and the other doesn't": {
+			buffersRegistry: NewCapacityBuffersFakePodsRegistry(map[string]*v1alpha1.CapacityBuffer{"fake-pod-1": buffer1, "fake-pod-2": buffer1}),
+			state: &status.ScaleUpStatus{
+				Result:               status.ScaleUpNoOptionsAvailable,
+				ScaleUpInfos:         []nodegroupset.ScaleUpInfo{{}},
+				PodsTriggeredScaleUp: []*apiv1.Pod{createPod("pod-1", false), createPod("fake-pod-1", true)},
+				PodsRemainUnschedulable: []status.NoScaleUpInfo{
+					{
+						Pod:                createPod("fake-pod-2", true),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+				},
+				PodsAwaitEvaluation: []*apiv1.Pod{},
+			},
+			expectedTriggeredScaleUp:    1,
+			expectedNotTriggeredScaleUp: 1,
+		},
+		"multiple pods for multiple buffers with mixed conditions": {
+			buffersRegistry: NewCapacityBuffersFakePodsRegistry(map[string]*v1alpha1.CapacityBuffer{
+				"fake-pod-1": buffer1,
+				"fake-pod-2": buffer1,
+				"fake-pod-3": buffer2,
+				"fake-pod-4": buffer2,
+			}),
+			state: &status.ScaleUpStatus{
+				Result:       status.ScaleUpNoOptionsAvailable,
+				ScaleUpInfos: []nodegroupset.ScaleUpInfo{{}},
+				PodsTriggeredScaleUp: []*apiv1.Pod{
+					createPod("pod-1", false),
+					createPod("fake-pod-2", true),
+					createPod("fake-pod-4", true),
+				},
+				PodsRemainUnschedulable: []status.NoScaleUpInfo{
+					{
+						Pod:                createPod("fake-pod-1", true),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+					{
+						Pod:                createPod("fake-pod-3", true),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+					{
+						Pod:                createPod("pod-2", false),
+						RejectedNodeGroups: reasons,
+						SkippedNodeGroups:  reasons,
+					},
+				},
+				PodsAwaitEvaluation: []*apiv1.Pod{},
+			},
+			expectedTriggeredScaleUp:    2,
```

Comment:
WDYT about changing this from int to a slice of error strings so that it is possible to validate the message content as well?